Bio/SeqFeature.py

```diff
@@ -162,9 +160,9 @@ def __init__(
             # TODO - Deprecation warning
             self.strand = strand
         self.id = id
-        if qualifiers is None:
-            qualifiers = OrderedDict()
-        self.qualifiers = qualifiers
+        self.qualifiers = {}
+        if qualifiers is not None:
+            self.qualifiers.update(qualifiers)
```

Comment:
Won't the update be slower? Honestly not checked but I would expect that.

Reply:
I see - add a comment then that this also checks dict-like input

Reply:
This was (one of) the reasons for #3601. I can add a comment, but ultimately, there should be consistent checks. It would also be an opportunity to improve the error messages for erroneous input.